Sources/AsyncHTTPClient/FoundationExtensions.swift

```diff
@@ -54,3 +54,16 @@ extension HTTPClient.Cookie {
         )
     }
 }
+
+extension HTTPClient.Body {
+    /// Create and stream body using `Data`.
+    ///
+    /// - parameters:
+    ///     - bytes: Body `Data` representation.
+    public static func data(_ bytes: Data) -> HTTPClient.Body {
+        func forwardToGenericImpl<C>(_ value: C) -> HTTPClient.Body where C: RandomAccessCollection, C.Element == UInt8 {
```

Comment:
@Lukasa do you know if the compiler will specialise this local function?

Reply:
It appears to: https://swift.godbolt.org/z/WKeeG5r7v

Reply:
As @karwa says, yes, the compiler should absolutely specialise this function. However, it's a pretty ugly thunk to have to have, so I'd like to ask whether we can differentiate between this method and the other method called `data` by renaming the other one. That way we don't need to write this thunk out explicitly.

Reply:
Hmm, well the [async API](https://github.com/swift-server/async-http-client/blob/972bcddedc1e7a4333c77fa4445a9e2798ff6a9f/Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift#L89) seems to use `bytes` for this. So we could use that.

I went with `data` to try and remove the Foundation.Data dependency whilst still supporting Data. But if we have to add the extension for source compatibility, that reasoning isn't quite as strong. Then again, there is value in having the same thing be spelled the same way, and it's the API we have, so... ⚖️ . The thunk is not going to be winning any beauty awards, for sure, but is it less ugly than having two spellings for essentially the same operation? I'm not sure.

If you really think it's worth it, I could rename it to `bytes`.

Reply:
In general I'm disinclined to rely on type-based method overloads. We've had a bunch of problems with this in the past where having methods whose name is the same that differ only based on type leans excessively heavily on the type checker. If these differed only on return value I'd outright refuse to accept them having the same name: differing based on argument type is less bad, but in this instance I think I'd still rather use a different name. There's existing prior art in NIO for using the word `data` in method names to mean "accepts/returns `Foundation.Data`" and `bytes` to mean "accepts/returns something else", so I think we're in good company here.

Reply:
Oh, I wasn't aware that pattern was used elsewhere in NIO. I've changed it to `.bytes` now.